src/python/pants/backend/go/util_rules/go_mod.py

```diff
@@ -74,18 +74,24 @@ class GoModInfo:
 
 @dataclass(frozen=True)
 class GoModInfoRequest(EngineAwareParameter):
-    address: Address
+    source: Address | GoModSourcesField
```

Comment:
I know you probably want to keep changes to a minimum, but it would be good to fix this to always be `GoModSourcesField`, even if that's in a followup.

Reply:
Three or four other callers use this API without already having the target in hand, so this was to reduce their overhead.